tests/unit/s2n_self_talk_nonblocking_test.c

```diff
@@ -58,19 +65,25 @@ int mock_client(struct s2n_test_io_pair *io_pair, uint8_t *expected_data, uint32
 
     /* Receive 10MB of data */
     uint32_t remaining = size;
-    while(remaining) {
+    while(remaining && (start_time - time(0)) < max_wait_time) {
         int r = s2n_recv(client_conn, ptr, remaining, &blocked);
+        TEST_DEBUG_PRINT("Client read: %d bytes\n", r);
         if (r < 0) {
             return 1;
         }
         remaining -= r;
         ptr += r;
+        if (size - remaining > 32000 && should_block) {
```

Comment:
"3200" needs to be a named constant. Where does that number come from?

Reply:
> "3200" needs to be a named constant. Where does that number come from?

It doesn't need to be anything in particular. I'll make it a percentage of the payload size, that way we only have one magic number.